Patch-Source: https://github.com/postgres/postgres/commit/b28b9b19bbe3410da4a805ef775e0383a66af314
--
From b28b9b19bbe3410da4a805ef775e0383a66af314 Mon Sep 17 00:00:00 2001
From: Tom Lane <tgl@sss.pgh.pa.us>
Date: Sat, 14 Sep 2024 17:55:03 -0400
Subject: [PATCH] Run regression tests with timezone America/Los_Angeles.

Historically we've used timezone "PST8PDT", but the recent release
2024b of tzdb changes the definition of that zone in a way that
breaks many test cases concerned with dates before 1970.  Although
we've not yet adopted 2024b into our own tree, this is already
problematic for people using --with-system-tzdata if their platform
has already adopted 2024b.  To work with both older and newer
versions of tzdb, switch to using "America/Los_Angeles", accepting
the ensuing changes in regression test results.

Back-patch to all supported branches.

Per report and patch from Wolfgang Walther.

Discussion: https://postgr.es/m/0a997455-5aba-4cf2-a354-d26d8bcbfae6@technowledgy.de
---
 doc/src/sgml/ref/set.sgml                 |   4 +-
 doc/src/sgml/regress.sgml                 |   6 +-
 src/test/regress/expected/date.out        |  10 +-
 src/test/regress/expected/horology.out    |  57 ++---
 src/test/regress/expected/timestamptz.out | 249 +++++++++++-----------
 src/test/regress/pg_regress.c             |   2 +-
 src/test/regress/sql/horology.sql         |   4 +-
 src/test/regress/sql/timestamptz.sql      |   5 +-
 8 files changed, 176 insertions(+), 161 deletions(-)

diff --git a/doc/src/sgml/ref/set.sgml b/doc/src/sgml/ref/set.sgml
index 9a7b387b29fed..20ebc122bf780 100644
--- a/doc/src/sgml/ref/set.sgml
+++ b/doc/src/sgml/ref/set.sgml
@@ -198,7 +198,7 @@ SELECT setseed(<replaceable>value</replaceable>);
 
        <variablelist>
         <varlistentry>
-         <term><literal>'PST8PDT'</literal></term>
+         <term><literal>'America/Los_Angeles'</literal></term>
          <listitem>
           <para>
            The time zone for Berkeley, California.
@@ -296,7 +296,7 @@ SET datestyle TO postgres, dmy;
   <para>
    Set the time zone for Berkeley, California:
 <screen>
-SET TIME ZONE 'PST8PDT';
+SET TIME ZONE 'America/Los_Angeles';
 </screen>
   </para>
 
diff --git a/doc/src/sgml/regress.sgml b/doc/src/sgml/regress.sgml
index 2fa9aea8d31f1..130be794da6d7 100644
--- a/doc/src/sgml/regress.sgml
+++ b/doc/src/sgml/regress.sgml
@@ -530,11 +530,11 @@ make check NO_LOCALE=1
     <para>
      Most of the date and time results are dependent on the time zone
      environment.  The reference files are generated for time zone
-     <literal>PST8PDT</literal> (Berkeley, California), and there will be
+     <literal>America/Los_Angeles</literal>, and there will be
      apparent failures if the tests are not run with that time zone setting.
      The regression test driver sets environment variable
-     <envar>PGTZ</envar> to <literal>PST8PDT</literal>, which normally
-     ensures proper results.
+     <envar>PGTZ</envar> to <literal>America/Los_Angeles</literal>,
+     which normally ensures proper results.
     </para>
    </sect2>
 
diff --git a/src/test/regress/expected/date.out b/src/test/regress/expected/date.out
index 4686d0d8cab46..7d753654f899d 100644
--- a/src/test/regress/expected/date.out
+++ b/src/test/regress/expected/date.out
@@ -1148,7 +1148,7 @@ SELECT DATE_TRUNC('MILLENNIUM', TIMESTAMP '1970-03-20 04:30:00.00000'); -- 1001
 SELECT DATE_TRUNC('MILLENNIUM', DATE '1970-03-20'); -- 1001-01-01
           date_trunc          
 ------------------------------
- Thu Jan 01 00:00:00 1001 PST
+ Thu Jan 01 00:00:00 1001 LMT
 (1 row)
 
 SELECT DATE_TRUNC('CENTURY', TIMESTAMP '1970-03-20 04:30:00.00000'); -- 1901
@@ -1172,13 +1172,13 @@ SELECT DATE_TRUNC('CENTURY', DATE '2004-08-10'); -- 2001-01-01
 SELECT DATE_TRUNC('CENTURY', DATE '0002-02-04'); -- 0001-01-01
           date_trunc          
 ------------------------------
- Mon Jan 01 00:00:00 0001 PST
+ Mon Jan 01 00:00:00 0001 LMT
 (1 row)
 
 SELECT DATE_TRUNC('CENTURY', DATE '0055-08-10 BC'); -- 0100-01-01 BC
            date_trunc            
 ---------------------------------
- Tue Jan 01 00:00:00 0100 PST BC
+ Tue Jan 01 00:00:00 0100 LMT BC
 (1 row)
 
 SELECT DATE_TRUNC('DECADE', DATE '1993-12-25'); -- 1990-01-01
@@ -1190,13 +1190,13 @@ SELECT DATE_TRUNC('DECADE', DATE '1993-12-25'); -- 1990-01-01
 SELECT DATE_TRUNC('DECADE', DATE '0004-12-25'); -- 0001-01-01 BC
            date_trunc            
 ---------------------------------
- Sat Jan 01 00:00:00 0001 PST BC
+ Sat Jan 01 00:00:00 0001 LMT BC
 (1 row)
 
 SELECT DATE_TRUNC('DECADE', DATE '0002-12-31 BC'); -- 0011-01-01 BC
            date_trunc            
 ---------------------------------
- Mon Jan 01 00:00:00 0011 PST BC
+ Mon Jan 01 00:00:00 0011 LMT BC
 (1 row)
 
 --
diff --git a/src/test/regress/expected/horology.out b/src/test/regress/expected/horology.out
index 8ae0f8ce61d15..ae447b8a3a81b 100644
--- a/src/test/regress/expected/horology.out
+++ b/src/test/regress/expected/horology.out
@@ -1,11 +1,16 @@
 --
 -- HOROLOGY
 --
-SET DateStyle = 'Postgres, MDY';
-SHOW TimeZone;  -- Many of these tests depend on the prevailing setting
- TimeZone 
-----------
- PST8PDT
+SHOW TimeZone;  -- Many of these tests depend on the prevailing settings
+      TimeZone       
+---------------------
+ America/Los_Angeles
+(1 row)
+
+SHOW DateStyle;
+   DateStyle   
+---------------
+ Postgres, MDY
 (1 row)
 
 --
@@ -765,12 +770,12 @@ SELECT '' AS "64", d1 + interval '1 year' AS one_year FROM TIMESTAMPTZ_TBL;
     | Sat Feb 14 17:32:01 1998 PST
     | Sun Feb 15 17:32:01 1998 PST
     | Mon Feb 16 17:32:01 1998 PST
-    | Thu Feb 16 17:32:01 0096 PST BC
-    | Sun Feb 16 17:32:01 0098 PST
-    | Fri Feb 16 17:32:01 0598 PST
-    | Wed Feb 16 17:32:01 1098 PST
-    | Sun Feb 16 17:32:01 1698 PST
-    | Fri Feb 16 17:32:01 1798 PST
+    | Thu Feb 16 17:32:01 0096 LMT BC
+    | Sun Feb 16 17:32:01 0098 LMT
+    | Fri Feb 16 17:32:01 0598 LMT
+    | Wed Feb 16 17:32:01 1098 LMT
+    | Sun Feb 16 17:32:01 1698 LMT
+    | Fri Feb 16 17:32:01 1798 LMT
     | Wed Feb 16 17:32:01 1898 PST
     | Mon Feb 16 17:32:01 1998 PST
     | Sun Feb 16 17:32:01 2098 PST
@@ -836,12 +841,12 @@ SELECT '' AS "64", d1 - interval '1 year' AS one_year FROM TIMESTAMPTZ_TBL;
     | Wed Feb 14 17:32:01 1996 PST
     | Thu Feb 15 17:32:01 1996 PST
     | Fri Feb 16 17:32:01 1996 PST
-    | Mon Feb 16 17:32:01 0098 PST BC
-    | Thu Feb 16 17:32:01 0096 PST
-    | Tue Feb 16 17:32:01 0596 PST
-    | Sun Feb 16 17:32:01 1096 PST
-    | Thu Feb 16 17:32:01 1696 PST
-    | Tue Feb 16 17:32:01 1796 PST
+    | Mon Feb 16 17:32:01 0098 LMT BC
+    | Thu Feb 16 17:32:01 0096 LMT
+    | Tue Feb 16 17:32:01 0596 LMT
+    | Sun Feb 16 17:32:01 1096 LMT
+    | Thu Feb 16 17:32:01 1696 LMT
+    | Tue Feb 16 17:32:01 1796 LMT
     | Sun Feb 16 17:32:01 1896 PST
     | Fri Feb 16 17:32:01 1996 PST
     | Thu Feb 16 17:32:01 2096 PST
@@ -2120,7 +2125,7 @@ SELECT '2020-10-05'::timestamptz > '2202020-10-05'::date as f;
 SELECT '4714-11-24 BC'::date::timestamptz;
            timestamptz           
 ---------------------------------
- Mon Nov 24 00:00:00 4714 PST BC
+ Mon Nov 24 00:00:00 4714 LMT BC
 (1 row)
 
 SET TimeZone = 'UTC-2';
@@ -2627,13 +2632,13 @@ RESET DateStyle;
 SELECT to_timestamp('0097/Feb/16 --> 08:14:30', 'YYYY/Mon/DD --> HH:MI:SS');
          to_timestamp         
 ------------------------------
- Sat Feb 16 08:14:30 0097 PST
+ Sat Feb 16 08:14:30 0097 LMT
 (1 row)
 
 SELECT to_timestamp('97/2/16 8:14:30', 'FMYYYY/FMMM/FMDD FMHH:FMMI:FMSS');
          to_timestamp         
 ------------------------------
- Sat Feb 16 08:14:30 0097 PST
+ Sat Feb 16 08:14:30 0097 LMT
 (1 row)
 
 SELECT to_timestamp('2011$03!18 23_38_15', 'YYYY-MM-DD HH24:MI:SS');
@@ -2670,7 +2675,7 @@ SELECT to_timestamp('My birthday-> Year: 1976, Month: May, Day: 16',
 SELECT to_timestamp('1,582nd VIII 21', 'Y,YYYth FMRM DD');
          to_timestamp         
 ------------------------------
- Sat Aug 21 00:00:00 1582 PST
+ Sat Aug 21 00:00:00 1582 LMT
 (1 row)
 
 SELECT to_timestamp('15 "text between quote marks" 98 54 45',
@@ -2734,7 +2739,7 @@ SELECT to_timestamp('1997 AD 11 16', 'YYYY BC MM DD');
 SELECT to_timestamp('1997 BC 11 16', 'YYYY BC MM DD');
           to_timestamp           
 ---------------------------------
- Tue Nov 16 00:00:00 1997 PST BC
+ Tue Nov 16 00:00:00 1997 LMT BC
 (1 row)
 
 SELECT to_timestamp('1997 A.D. 11 16', 'YYYY B.C. MM DD');
@@ -2746,7 +2751,7 @@ SELECT to_timestamp('1997 A.D. 11 16', 'YYYY B.C. MM DD');
 SELECT to_timestamp('1997 B.C. 11 16', 'YYYY B.C. MM DD');
           to_timestamp           
 ---------------------------------
- Tue Nov 16 00:00:00 1997 PST BC
+ Tue Nov 16 00:00:00 1997 LMT BC
 (1 row)
 
 SELECT to_timestamp('9-1116', 'Y-MMDD');
@@ -3016,19 +3021,19 @@ SELECT to_date('-44-02-01 BC','YYYY-MM-DD BC');
 SELECT to_timestamp('44-02-01 11:12:13 BC','YYYY-MM-DD HH24:MI:SS BC');
           to_timestamp           
 ---------------------------------
- Fri Feb 01 11:12:13 0044 PST BC
+ Fri Feb 01 11:12:13 0044 LMT BC
 (1 row)
 
 SELECT to_timestamp('-44-02-01 11:12:13','YYYY-MM-DD HH24:MI:SS');
           to_timestamp           
 ---------------------------------
- Fri Feb 01 11:12:13 0044 PST BC
+ Fri Feb 01 11:12:13 0044 LMT BC
 (1 row)
 
 SELECT to_timestamp('-44-02-01 11:12:13 BC','YYYY-MM-DD HH24:MI:SS BC');
          to_timestamp         
 ------------------------------
- Mon Feb 01 11:12:13 0044 PST
+ Mon Feb 01 11:12:13 0044 LMT
 (1 row)
 
 --
diff --git a/src/test/regress/expected/timestamptz.out b/src/test/regress/expected/timestamptz.out
index 295ef6ac251a3..0ce33862019b2 100644
--- a/src/test/regress/expected/timestamptz.out
+++ b/src/test/regress/expected/timestamptz.out
@@ -293,12 +293,12 @@ SELECT '' AS "64", d1 FROM TIMESTAMPTZ_TBL;
     | Fri Feb 14 17:32:01 1997 PST
     | Sat Feb 15 17:32:01 1997 PST
     | Sun Feb 16 17:32:01 1997 PST
-    | Tue Feb 16 17:32:01 0097 PST BC
-    | Sat Feb 16 17:32:01 0097 PST
-    | Thu Feb 16 17:32:01 0597 PST
-    | Tue Feb 16 17:32:01 1097 PST
-    | Sat Feb 16 17:32:01 1697 PST
-    | Thu Feb 16 17:32:01 1797 PST
+    | Tue Feb 16 17:32:01 0097 LMT BC
+    | Sat Feb 16 17:32:01 0097 LMT
+    | Thu Feb 16 17:32:01 0597 LMT
+    | Tue Feb 16 17:32:01 1097 LMT
+    | Sat Feb 16 17:32:01 1697 LMT
+    | Thu Feb 16 17:32:01 1797 LMT
     | Tue Feb 16 17:32:01 1897 PST
     | Sun Feb 16 17:32:01 1997 PST
     | Sat Feb 16 17:32:01 2097 PST
@@ -322,19 +322,19 @@ SELECT '' AS "64", d1 FROM TIMESTAMPTZ_TBL;
 SELECT '4714-11-24 00:00:00+00 BC'::timestamptz;
            timestamptz           
 ---------------------------------
- Sun Nov 23 16:00:00 4714 PST BC
+ Sun Nov 23 16:07:02 4714 LMT BC
 (1 row)
 
 SELECT '4714-11-23 16:00:00-08 BC'::timestamptz;
            timestamptz           
 ---------------------------------
- Sun Nov 23 16:00:00 4714 PST BC
+ Sun Nov 23 16:07:02 4714 LMT BC
 (1 row)
 
 SELECT 'Sun Nov 23 16:00:00 4714 PST BC'::timestamptz;
            timestamptz           
 ---------------------------------
- Sun Nov 23 16:00:00 4714 PST BC
+ Sun Nov 23 16:07:02 4714 LMT BC
 (1 row)
 
 SELECT '4714-11-23 23:59:59+00 BC'::timestamptz;  -- out of range
@@ -405,12 +405,12 @@ SELECT '' AS "15", d1 FROM TIMESTAMPTZ_TBL
 ----+---------------------------------
     | -infinity
     | Wed Dec 31 16:00:00 1969 PST
-    | Tue Feb 16 17:32:01 0097 PST BC
-    | Sat Feb 16 17:32:01 0097 PST
-    | Thu Feb 16 17:32:01 0597 PST
-    | Tue Feb 16 17:32:01 1097 PST
-    | Sat Feb 16 17:32:01 1697 PST
-    | Thu Feb 16 17:32:01 1797 PST
+    | Tue Feb 16 17:32:01 0097 LMT BC
+    | Sat Feb 16 17:32:01 0097 LMT
+    | Thu Feb 16 17:32:01 0597 LMT
+    | Tue Feb 16 17:32:01 1097 LMT
+    | Sat Feb 16 17:32:01 1697 LMT
+    | Thu Feb 16 17:32:01 1797 LMT
     | Tue Feb 16 17:32:01 1897 PST
     | Wed Feb 28 17:32:01 1996 PST
     | Thu Feb 29 17:32:01 1996 PST
@@ -473,12 +473,12 @@ SELECT '' AS "63", d1 FROM TIMESTAMPTZ_TBL
     | Fri Feb 14 17:32:01 1997 PST
     | Sat Feb 15 17:32:01 1997 PST
     | Sun Feb 16 17:32:01 1997 PST
-    | Tue Feb 16 17:32:01 0097 PST BC
-    | Sat Feb 16 17:32:01 0097 PST
-    | Thu Feb 16 17:32:01 0597 PST
-    | Tue Feb 16 17:32:01 1097 PST
-    | Sat Feb 16 17:32:01 1697 PST
-    | Thu Feb 16 17:32:01 1797 PST
+    | Tue Feb 16 17:32:01 0097 LMT BC
+    | Sat Feb 16 17:32:01 0097 LMT
+    | Thu Feb 16 17:32:01 0597 LMT
+    | Tue Feb 16 17:32:01 1097 LMT
+    | Sat Feb 16 17:32:01 1697 LMT
+    | Thu Feb 16 17:32:01 1797 LMT
     | Tue Feb 16 17:32:01 1897 PST
     | Sun Feb 16 17:32:01 1997 PST
     | Sat Feb 16 17:32:01 2097 PST
@@ -505,12 +505,12 @@ SELECT '' AS "16", d1 FROM TIMESTAMPTZ_TBL
     | -infinity
     | Wed Dec 31 16:00:00 1969 PST
     | Thu Jan 02 00:00:00 1997 PST
-    | Tue Feb 16 17:32:01 0097 PST BC
-    | Sat Feb 16 17:32:01 0097 PST
-    | Thu Feb 16 17:32:01 0597 PST
-    | Tue Feb 16 17:32:01 1097 PST
-    | Sat Feb 16 17:32:01 1697 PST
-    | Thu Feb 16 17:32:01 1797 PST
+    | Tue Feb 16 17:32:01 0097 LMT BC
+    | Sat Feb 16 17:32:01 0097 LMT
+    | Thu Feb 16 17:32:01 0597 LMT
+    | Tue Feb 16 17:32:01 1097 LMT
+    | Sat Feb 16 17:32:01 1697 LMT
+    | Thu Feb 16 17:32:01 1797 LMT
     | Tue Feb 16 17:32:01 1897 PST
     | Wed Feb 28 17:32:01 1996 PST
     | Thu Feb 29 17:32:01 1996 PST
@@ -778,12 +778,12 @@ SELECT d1 as timestamptz,
  Fri Feb 14 17:32:01 1997 PST    |      1997 |     2 |  14 |   17 |     32 |      1
  Sat Feb 15 17:32:01 1997 PST    |      1997 |     2 |  15 |   17 |     32 |      1
  Sun Feb 16 17:32:01 1997 PST    |      1997 |     2 |  16 |   17 |     32 |      1
- Tue Feb 16 17:32:01 0097 PST BC |       -97 |     2 |  16 |   17 |     32 |      1
- Sat Feb 16 17:32:01 0097 PST    |        97 |     2 |  16 |   17 |     32 |      1
- Thu Feb 16 17:32:01 0597 PST    |       597 |     2 |  16 |   17 |     32 |      1
- Tue Feb 16 17:32:01 1097 PST    |      1097 |     2 |  16 |   17 |     32 |      1
- Sat Feb 16 17:32:01 1697 PST    |      1697 |     2 |  16 |   17 |     32 |      1
- Thu Feb 16 17:32:01 1797 PST    |      1797 |     2 |  16 |   17 |     32 |      1
+ Tue Feb 16 17:32:01 0097 LMT BC |       -97 |     2 |  16 |   17 |     32 |      1
+ Sat Feb 16 17:32:01 0097 LMT    |        97 |     2 |  16 |   17 |     32 |      1
+ Thu Feb 16 17:32:01 0597 LMT    |       597 |     2 |  16 |   17 |     32 |      1
+ Tue Feb 16 17:32:01 1097 LMT    |      1097 |     2 |  16 |   17 |     32 |      1
+ Sat Feb 16 17:32:01 1697 LMT    |      1697 |     2 |  16 |   17 |     32 |      1
+ Thu Feb 16 17:32:01 1797 LMT    |      1797 |     2 |  16 |   17 |     32 |      1
  Tue Feb 16 17:32:01 1897 PST    |      1897 |     2 |  16 |   17 |     32 |      1
  Sun Feb 16 17:32:01 1997 PST    |      1997 |     2 |  16 |   17 |     32 |      1
  Sat Feb 16 17:32:01 2097 PST    |      2097 |     2 |  16 |   17 |     32 |      1
@@ -852,12 +852,12 @@ SELECT d1 as timestamptz,
  Fri Feb 14 17:32:01 1997 PST    |       1 |  1000 |  1000000
  Sat Feb 15 17:32:01 1997 PST    |       1 |  1000 |  1000000
  Sun Feb 16 17:32:01 1997 PST    |       1 |  1000 |  1000000
- Tue Feb 16 17:32:01 0097 PST BC |       1 |  1000 |  1000000
- Sat Feb 16 17:32:01 0097 PST    |       1 |  1000 |  1000000
- Thu Feb 16 17:32:01 0597 PST    |       1 |  1000 |  1000000
- Tue Feb 16 17:32:01 1097 PST    |       1 |  1000 |  1000000
- Sat Feb 16 17:32:01 1697 PST    |       1 |  1000 |  1000000
- Thu Feb 16 17:32:01 1797 PST    |       1 |  1000 |  1000000
+ Tue Feb 16 17:32:01 0097 LMT BC |       1 |  1000 |  1000000
+ Sat Feb 16 17:32:01 0097 LMT    |       1 |  1000 |  1000000
+ Thu Feb 16 17:32:01 0597 LMT    |       1 |  1000 |  1000000
+ Tue Feb 16 17:32:01 1097 LMT    |       1 |  1000 |  1000000
+ Sat Feb 16 17:32:01 1697 LMT    |       1 |  1000 |  1000000
+ Thu Feb 16 17:32:01 1797 LMT    |       1 |  1000 |  1000000
  Tue Feb 16 17:32:01 1897 PST    |       1 |  1000 |  1000000
  Sun Feb 16 17:32:01 1997 PST    |       1 |  1000 |  1000000
  Sat Feb 16 17:32:01 2097 PST    |       1 |  1000 |  1000000
@@ -927,12 +927,12 @@ SELECT d1 as timestamptz,
  Fri Feb 14 17:32:01 1997 PST    |      1997 |    7 |      5 |   5 |  45
  Sat Feb 15 17:32:01 1997 PST    |      1997 |    7 |      6 |   6 |  46
  Sun Feb 16 17:32:01 1997 PST    |      1997 |    7 |      7 |   0 |  47
- Tue Feb 16 17:32:01 0097 PST BC |       -97 |    7 |      2 |   2 |  47
- Sat Feb 16 17:32:01 0097 PST    |        97 |    7 |      6 |   6 |  47
- Thu Feb 16 17:32:01 0597 PST    |       597 |    7 |      4 |   4 |  47
- Tue Feb 16 17:32:01 1097 PST    |      1097 |    7 |      2 |   2 |  47
- Sat Feb 16 17:32:01 1697 PST    |      1697 |    7 |      6 |   6 |  47
- Thu Feb 16 17:32:01 1797 PST    |      1797 |    7 |      4 |   4 |  47
+ Tue Feb 16 17:32:01 0097 LMT BC |       -97 |    7 |      2 |   2 |  47
+ Sat Feb 16 17:32:01 0097 LMT    |        97 |    7 |      6 |   6 |  47
+ Thu Feb 16 17:32:01 0597 LMT    |       597 |    7 |      4 |   4 |  47
+ Tue Feb 16 17:32:01 1097 LMT    |      1097 |    7 |      2 |   2 |  47
+ Sat Feb 16 17:32:01 1697 LMT    |      1697 |    7 |      6 |   6 |  47
+ Thu Feb 16 17:32:01 1797 LMT    |      1797 |    7 |      4 |   4 |  47
  Tue Feb 16 17:32:01 1897 PST    |      1897 |    7 |      2 |   2 |  47
  Sun Feb 16 17:32:01 1997 PST    |      1997 |    7 |      7 |   0 |  47
  Sat Feb 16 17:32:01 2097 PST    |      2097 |    7 |      6 |   6 |  47
@@ -1003,12 +1003,12 @@ SELECT d1 as timestamptz,
  Fri Feb 14 17:32:01 1997 PST    |       199 |        20 |          2 |   2450495
  Sat Feb 15 17:32:01 1997 PST    |       199 |        20 |          2 |   2450496
  Sun Feb 16 17:32:01 1997 PST    |       199 |        20 |          2 |   2450497
- Tue Feb 16 17:32:01 0097 PST BC |       -10 |        -1 |         -1 |   1686043
- Sat Feb 16 17:32:01 0097 PST    |         9 |         1 |          1 |   1756537
- Thu Feb 16 17:32:01 0597 PST    |        59 |         6 |          1 |   1939158
- Tue Feb 16 17:32:01 1097 PST    |       109 |        11 |          2 |   2121779
- Sat Feb 16 17:32:01 1697 PST    |       169 |        17 |          2 |   2340925
- Thu Feb 16 17:32:01 1797 PST    |       179 |        18 |          2 |   2377449
+ Tue Feb 16 17:32:01 0097 LMT BC |       -10 |        -1 |         -1 |   1686043
+ Sat Feb 16 17:32:01 0097 LMT    |         9 |         1 |          1 |   1756537
+ Thu Feb 16 17:32:01 0597 LMT    |        59 |         6 |          1 |   1939158
+ Tue Feb 16 17:32:01 1097 LMT    |       109 |        11 |          2 |   2121779
+ Sat Feb 16 17:32:01 1697 LMT    |       169 |        17 |          2 |   2340925
+ Thu Feb 16 17:32:01 1797 LMT    |       179 |        18 |          2 |   2377449
  Tue Feb 16 17:32:01 1897 PST    |       189 |        19 |          2 |   2413973
  Sun Feb 16 17:32:01 1997 PST    |       199 |        20 |          2 |   2450497
  Sat Feb 16 17:32:01 2097 PST    |       209 |        21 |          3 |   2487022
@@ -1033,74 +1033,74 @@ SELECT d1 as timestamptz,
    date_part( 'timezone_hour', d1) AS timezone_hour,
    date_part( 'timezone_minute', d1) AS timezone_minute
    FROM TIMESTAMPTZ_TBL;
-           timestamptz           | timezone | timezone_hour | timezone_minute 
----------------------------------+----------+---------------+-----------------
- -infinity                       |          |               |                
- infinity                        |          |               |                
- Wed Dec 31 16:00:00 1969 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:02 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01.4 1997 PST  |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01.5 1997 PST  |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01.6 1997 PST  |   -28800 |            -8 |               0
- Thu Jan 02 00:00:00 1997 PST    |   -28800 |            -8 |               0
- Thu Jan 02 03:04:05 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Tue Jun 10 17:32:01 1997 PDT    |   -25200 |            -7 |               0
- Sat Sep 22 18:19:20 2001 PDT    |   -25200 |            -7 |               0
- Wed Mar 15 08:14:01 2000 PST    |   -28800 |            -8 |               0
- Wed Mar 15 04:14:02 2000 PST    |   -28800 |            -8 |               0
- Wed Mar 15 02:14:03 2000 PST    |   -28800 |            -8 |               0
- Wed Mar 15 03:14:04 2000 PST    |   -28800 |            -8 |               0
- Wed Mar 15 01:14:05 2000 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:00 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 09:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 09:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 09:32:01 1997 PST    |   -28800 |            -8 |               0
- Mon Feb 10 14:32:01 1997 PST    |   -28800 |            -8 |               0
- Thu Jul 10 14:32:01 1997 PDT    |   -25200 |            -7 |               0
- Tue Jun 10 18:32:01 1997 PDT    |   -25200 |            -7 |               0
- Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Tue Feb 11 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Wed Feb 12 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Thu Feb 13 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Fri Feb 14 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Sat Feb 15 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Sun Feb 16 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Tue Feb 16 17:32:01 0097 PST BC |   -28800 |            -8 |               0
- Sat Feb 16 17:32:01 0097 PST    |   -28800 |            -8 |               0
- Thu Feb 16 17:32:01 0597 PST    |   -28800 |            -8 |               0
- Tue Feb 16 17:32:01 1097 PST    |   -28800 |            -8 |               0
- Sat Feb 16 17:32:01 1697 PST    |   -28800 |            -8 |               0
- Thu Feb 16 17:32:01 1797 PST    |   -28800 |            -8 |               0
- Tue Feb 16 17:32:01 1897 PST    |   -28800 |            -8 |               0
- Sun Feb 16 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Sat Feb 16 17:32:01 2097 PST    |   -28800 |            -8 |               0
- Wed Feb 28 17:32:01 1996 PST    |   -28800 |            -8 |               0
- Thu Feb 29 17:32:01 1996 PST    |   -28800 |            -8 |               0
- Fri Mar 01 17:32:01 1996 PST    |   -28800 |            -8 |               0
- Mon Dec 30 17:32:01 1996 PST    |   -28800 |            -8 |               0
- Tue Dec 31 17:32:01 1996 PST    |   -28800 |            -8 |               0
- Wed Jan 01 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Fri Feb 28 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Sat Mar 01 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Tue Dec 30 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Wed Dec 31 17:32:01 1997 PST    |   -28800 |            -8 |               0
- Fri Dec 31 17:32:01 1999 PST    |   -28800 |            -8 |               0
- Sat Jan 01 17:32:01 2000 PST    |   -28800 |            -8 |               0
- Sun Dec 31 17:32:01 2000 PST    |   -28800 |            -8 |               0
- Mon Jan 01 17:32:01 2001 PST    |   -28800 |            -8 |               0
+           timestamptz           | timezone | timezone_hour |  timezone_minute   
+---------------------------------+----------+---------------+--------------------
+ -infinity                       |          |               |                   
+ infinity                        |          |               |                   
+ Wed Dec 31 16:00:00 1969 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:02 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01.4 1997 PST  |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01.5 1997 PST  |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01.6 1997 PST  |   -28800 |            -8 |                  0
+ Thu Jan 02 00:00:00 1997 PST    |   -28800 |            -8 |                  0
+ Thu Jan 02 03:04:05 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Tue Jun 10 17:32:01 1997 PDT    |   -25200 |            -7 |                  0
+ Sat Sep 22 18:19:20 2001 PDT    |   -25200 |            -7 |                  0
+ Wed Mar 15 08:14:01 2000 PST    |   -28800 |            -8 |                  0
+ Wed Mar 15 04:14:02 2000 PST    |   -28800 |            -8 |                  0
+ Wed Mar 15 02:14:03 2000 PST    |   -28800 |            -8 |                  0
+ Wed Mar 15 03:14:04 2000 PST    |   -28800 |            -8 |                  0
+ Wed Mar 15 01:14:05 2000 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:00 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 09:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 09:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 09:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Mon Feb 10 14:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Thu Jul 10 14:32:01 1997 PDT    |   -25200 |            -7 |                  0
+ Tue Jun 10 18:32:01 1997 PDT    |   -25200 |            -7 |                  0
+ Mon Feb 10 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Tue Feb 11 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Wed Feb 12 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Thu Feb 13 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Fri Feb 14 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Sat Feb 15 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Sun Feb 16 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Tue Feb 16 17:32:01 0097 LMT BC |   -28378 |            -7 | -52.96666666666664
+ Sat Feb 16 17:32:01 0097 LMT    |   -28378 |            -7 | -52.96666666666664
+ Thu Feb 16 17:32:01 0597 LMT    |   -28378 |            -7 | -52.96666666666664
+ Tue Feb 16 17:32:01 1097 LMT    |   -28378 |            -7 | -52.96666666666664
+ Sat Feb 16 17:32:01 1697 LMT    |   -28378 |            -7 | -52.96666666666664
+ Thu Feb 16 17:32:01 1797 LMT    |   -28378 |            -7 | -52.96666666666664
+ Tue Feb 16 17:32:01 1897 PST    |   -28800 |            -8 |                  0
+ Sun Feb 16 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Sat Feb 16 17:32:01 2097 PST    |   -28800 |            -8 |                  0
+ Wed Feb 28 17:32:01 1996 PST    |   -28800 |            -8 |                  0
+ Thu Feb 29 17:32:01 1996 PST    |   -28800 |            -8 |                  0
+ Fri Mar 01 17:32:01 1996 PST    |   -28800 |            -8 |                  0
+ Mon Dec 30 17:32:01 1996 PST    |   -28800 |            -8 |                  0
+ Tue Dec 31 17:32:01 1996 PST    |   -28800 |            -8 |                  0
+ Wed Jan 01 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Fri Feb 28 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Sat Mar 01 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Tue Dec 30 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Wed Dec 31 17:32:01 1997 PST    |   -28800 |            -8 |                  0
+ Fri Dec 31 17:32:01 1999 PST    |   -28800 |            -8 |                  0
+ Sat Jan 01 17:32:01 2000 PST    |   -28800 |            -8 |                  0
+ Sun Dec 31 17:32:01 2000 PST    |   -28800 |            -8 |                  0
+ Mon Jan 01 17:32:01 2001 PST    |   -28800 |            -8 |                  0
 (66 rows)
 
 -- TO_CHAR()
@@ -1988,7 +1988,7 @@ INSERT INTO TIMESTAMPTZ_TST VALUES(4, '1000000312 23:58:48 IST');
 SELECT * FROM TIMESTAMPTZ_TST ORDER BY a;
  a |               b                
 ---+--------------------------------
- 1 | Wed Mar 12 13:58:48 1000 PST
+ 1 | Wed Mar 12 14:05:50 1000 LMT
  2 | Sun Mar 12 14:58:48 10000 PDT
  3 | Sun Mar 12 14:58:48 100000 PDT
  3 | Sun Mar 12 14:58:48 10000 PDT
@@ -2099,7 +2099,14 @@ SELECT make_timestamptz(2008, 12, 10, 10, 10, 10, 'EDT');
  Wed Dec 10 09:10:10 2008 EST
 (1 row)
 
-SELECT make_timestamptz(2014, 12, 10, 10, 10, 10, 'PST8PDT');
+SELECT make_timestamptz(2014, 12, 10, 10, 10, 10, 'FOO8BAR');
+       make_timestamptz       
+------------------------------
+ Wed Dec 10 13:10:10 2014 EST
+(1 row)
+
+-- POSIX
+SELECT make_timestamptz(2014, 12, 10, 10, 10, 10, 'PST8PDT,M3.2.0,M11.1.0');
        make_timestamptz       
 ------------------------------
  Wed Dec 10 13:10:10 2014 EST
diff --git a/src/test/regress/pg_regress.c b/src/test/regress/pg_regress.c
index 1085034276072..d87f514cd8e63 100644
--- a/src/test/regress/pg_regress.c
+++ b/src/test/regress/pg_regress.c
@@ -777,7 +777,7 @@ initialize_environment(void)
 	/*
 	 * Set timezone and datestyle for datetime-related tests
 	 */
-	putenv("PGTZ=PST8PDT");
+	putenv("PGTZ=America/Los_Angeles");
 	putenv("PGDATESTYLE=Postgres, MDY");
 
 	/*
diff --git a/src/test/regress/sql/horology.sql b/src/test/regress/sql/horology.sql
index 53fabed8af31a..4ab5c1dfc8fd0 100644
--- a/src/test/regress/sql/horology.sql
+++ b/src/test/regress/sql/horology.sql
@@ -1,9 +1,9 @@
 --
 -- HOROLOGY
 --
-SET DateStyle = 'Postgres, MDY';
 
-SHOW TimeZone;  -- Many of these tests depend on the prevailing setting
+SHOW TimeZone;  -- Many of these tests depend on the prevailing settings
+SHOW DateStyle;
 
 --
 -- Test various input formats
diff --git a/src/test/regress/sql/timestamptz.sql b/src/test/regress/sql/timestamptz.sql
index 8e7d624ec0973..f0a8d3e5aa6dc 100644
--- a/src/test/regress/sql/timestamptz.sql
+++ b/src/test/regress/sql/timestamptz.sql
@@ -338,7 +338,10 @@ SELECT make_timestamptz(1910, 12, 24, 0, 0, 0, 'Nehwon/Lankhmar');
 -- abbreviations
 SELECT make_timestamptz(2008, 12, 10, 10, 10, 10, 'EST');
 SELECT make_timestamptz(2008, 12, 10, 10, 10, 10, 'EDT');
-SELECT make_timestamptz(2014, 12, 10, 10, 10, 10, 'PST8PDT');
+SELECT make_timestamptz(2014, 12, 10, 10, 10, 10, 'FOO8BAR');
+
+-- POSIX
+SELECT make_timestamptz(2014, 12, 10, 10, 10, 10, 'PST8PDT,M3.2.0,M11.1.0');
 
 RESET TimeZone;
 
